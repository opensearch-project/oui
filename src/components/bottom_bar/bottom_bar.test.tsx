/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { render, mount } from 'enzyme';
import { keysOf } from '../common';
import { requiredProps, takeMountedSnapshot } from '../../test';

import {
  OuiBottomBar,
  paddingSizeToClassNameMap,
  POSITIONS,
} from './bottom_bar';

// @ts-ignore TODO: Temporary hack which we can remove once react-test-renderer supports portals.
// More info at https://github.com/facebook/react/issues/11565.
ReactDOM.createPortal = (children) => {
  // hack to make enzyme treat the portal as a fragment
  if (children == null) return [['nested']];
  return children;
};

describe('OuiBottomBar', () => {
  test('is rendered', () => {
    const component = render(
      <OuiBottomBar {...requiredProps}>Content</OuiBottomBar>
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    describe('paddingSize', () => {
      keysOf(paddingSizeToClassNameMap).forEach((paddingSize) => {
        test(`${paddingSize} is rendered`, () => {
          const component = render(<OuiBottomBar paddingSize={paddingSize} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('position', () => {
      POSITIONS.forEach((position) => {
        test(`${position} is rendered`, () => {
          const component = render(<OuiBottomBar position={position} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    test('landmarkHeading', () => {
      const component = render(
        <OuiBottomBar landmarkHeading="This should have been label" />
      );

      expect(component).toMatchSnapshot();
    });

    test('affordForDisplacement can be false', () => {
      const component = render(<OuiBottomBar affordForDisplacement={false} />);

      expect(component).toMatchSnapshot();
    });

    test('insert root prop is altered', () => {
      const component = render(
        <OuiBottomBar insert={{ root: document.body }} />
      );

      expect(component).toMatchSnapshot();
    });

    test('insert sibling prop is altered', () => {
      const component = render(
        <OuiBottomBar insert={{ sibling: document.body, position: 'after' }} />
      );

      expect(component).toMatchSnapshot();
    });

    test('bodyClassName is rendered', () => {
      const component = mount(<OuiBottomBar bodyClassName={'customClass'} />);

      expect(takeMountedSnapshot(component)).toMatchSnapshot();
      expect(document.body.classList.contains('customClass')).toBe(true);
    });

    test('style is customized', () => {
      const component = render(<OuiBottomBar style={{ left: 12 }} />);

      expect(component).toMatchSnapshot();
    });

    test('position props are altered', () => {
      const component = render(
        <OuiBottomBar top={30} right={30} bottom={30} left={30} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
