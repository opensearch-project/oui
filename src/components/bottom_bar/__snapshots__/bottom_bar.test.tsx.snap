// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiBottomBar is rendered 1`] = `
Array [
  <section
    aria-label="aria-label"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium testClass1 testClass2"
    data-test-subj="test subject string"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
    Content
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props affordForDisplacement can be false 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props bodyClassName is rendered 1`] = `
<section
  aria-label="Page level controls"
  class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
>
  <h2
    class="ouiScreenReaderOnly"
  >
    Page level controls
  </h2>
</section>
`;

exports[`OuiBottomBar props landmarkHeading 1`] = `
Array [
  <section
    aria-label="This should have been label"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      This should have been label
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark called This should have been label with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props paddingSize l is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingLarge"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props paddingSize m is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props paddingSize none is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props paddingSize s is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingSmall"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props position fixed is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props position props are altered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
    style="left:30px;right:30px;bottom:30px;top:30px"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props position static is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--static ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props position sticky is rendered 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--sticky ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props style is customized 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
    style="left:12px"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;

exports[`OuiBottomBar props usePortal can be false 1`] = `
Array [
  <section
    aria-label="Page level controls"
    class="ouiBottomBar ouiBottomBar--fixed ouiBottomBar--paddingMedium"
  >
    <h2
      class="ouiScreenReaderOnly"
    >
      Page level controls
    </h2>
  </section>,
  <p
    aria-live="assertive"
    class="ouiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>,
]
`;
